nominal/core/asset.py

```diff
@@ -315,14 +199,19 @@ def unarchive(self) -> None:
         """Unarchive this asset, allowing it to be viewed in the UI."""
         self._clients.assets.unarchive(self._clients.auth_header, self.rid)
 
-    def _remove_data_sources(
+    def remove_data_scopes(
         self,
         *,
-        data_scope_names: Sequence[str] | None = None,
-        data_sources: Sequence[ScopeType | str] | None = None,
+        names: Sequence[str] | None = None,
+        scopes: Sequence[ScopeType | str] | None = None,
     ) -> None:
-        data_scope_names = data_scope_names or []
-        data_sources = data_sources or []
+        """Remove data scopes from this asset.
+
+        `names` are scope names.
+        `scopes` are rids or scope objects.
+        """
+        data_scope_names = set(names or [])
+        data_sources = set([rid_from_instance_or_string(scope) for scope in scopes] if scopes else [])
```

Comment:
I would prefer that we keep uniformity like we do elsewhere in the client of explicitly checking for None in a ternary and not utilizing as much short circuiting